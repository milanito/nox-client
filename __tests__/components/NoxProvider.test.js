import React from 'react'
import { mount } from 'enzyme'
import { forEach } from 'lodash'

import { NoxProvider } from '../../src'

describe('NoxProvider test cases', () => {
  describe('NoxProvider error cases', () => {
    forEach([{
      title: 'should require options',
      options: null
    }, {
      title: 'should not work with empty object options',
      options: {}
    }, {
      title: 'should not work with a wrong HTTP url',
      options: { baseURL: 'toto://toto.com' }
    }], ({ title, options }) =>
      it(title, () => {
        const wrapper = mount(
          <NoxProvider options={options}>
            <div className='unique' />
          </NoxProvider>
        )

        expect(wrapper.contains(<div className='unique' />)).toBeTruthy()
        expect(wrapper.props()).not.toHaveProperty('value')
        wrapper.unmount()
      }))
  })
  describe('NoxProvider success cases', () => {
    it('should render correctly', () => {
      const wrapper = mount(
        <NoxProvider options={{ baseURL: 'http://google.com' }}>
          <div className='unique' />
        </NoxProvider>
      )

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.contains(<div className='unique' />)).toBeTruthy()
      expect(wrapper.props()).toHaveProperty('options')
      wrapper.unmount()
    })
  })
})
