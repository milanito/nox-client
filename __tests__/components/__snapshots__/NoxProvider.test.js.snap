// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoxProvider test cases NoxProvider success cases should render correctly 1`] = `
<withState(withState(withHandlers(lifecycle(branch(Component)))))
  options={
    Object {
      "baseURL": "http://google.com",
    }
  }
>
  <withState(withHandlers(lifecycle(branch(Component))))
    error={null}
    modifyError={[Function]}
    options={
      Object {
        "baseURL": "http://google.com",
      }
    }
  >
    <withHandlers(lifecycle(branch(Component)))
      client={
        NoxClient {
          "cache": Cache {
            "clear": [Function],
            "debug": [Function],
            "del": [Function],
            "exportJson": [Function],
            "get": [Function],
            "hits": [Function],
            "importJson": [Function],
            "keys": [Function],
            "memsize": [Function],
            "misses": [Function],
            "put": [Function],
            "size": [Function],
          },
          "cacheTimeout": 600000,
          "client": [Function],
        }
      }
      error={null}
      modifyClient={[Function]}
      modifyError={[Function]}
      options={
        Object {
          "baseURL": "http://google.com",
        }
      }
    >
      <lifecycle(branch(Component))
        client={
          NoxClient {
            "cache": Cache {
              "clear": [Function],
              "debug": [Function],
              "del": [Function],
              "exportJson": [Function],
              "get": [Function],
              "hits": [Function],
              "importJson": [Function],
              "keys": [Function],
              "memsize": [Function],
              "misses": [Function],
              "put": [Function],
              "size": [Function],
            },
            "cacheTimeout": 600000,
            "client": [Function],
          }
        }
        error={null}
        modifyClient={[Function]}
        modifyError={[Function]}
        options={
          Object {
            "baseURL": "http://google.com",
          }
        }
        updateClient={[Function]}
        updateError={[Function]}
      >
        <branch(Component)
          client={
            NoxClient {
              "cache": Cache {
                "clear": [Function],
                "debug": [Function],
                "del": [Function],
                "exportJson": [Function],
                "get": [Function],
                "hits": [Function],
                "importJson": [Function],
                "keys": [Function],
                "memsize": [Function],
                "misses": [Function],
                "put": [Function],
                "size": [Function],
              },
              "cacheTimeout": 600000,
              "client": [Function],
            }
          }
          error={null}
          modifyClient={[Function]}
          modifyError={[Function]}
          options={
            Object {
              "baseURL": "http://google.com",
            }
          }
          updateClient={[Function]}
          updateError={[Function]}
        >
          <Component
            client={
              NoxClient {
                "cache": Cache {
                  "clear": [Function],
                  "debug": [Function],
                  "del": [Function],
                  "exportJson": [Function],
                  "get": [Function],
                  "hits": [Function],
                  "importJson": [Function],
                  "keys": [Function],
                  "memsize": [Function],
                  "misses": [Function],
                  "put": [Function],
                  "size": [Function],
                },
                "cacheTimeout": 600000,
                "client": [Function],
              }
            }
            error={null}
            modifyClient={[Function]}
            modifyError={[Function]}
            options={
              Object {
                "baseURL": "http://google.com",
              }
            }
            updateClient={[Function]}
            updateError={[Function]}
          >
            <div
              className="unique"
            />
          </Component>
        </branch(Component)>
      </lifecycle(branch(Component))>
    </withHandlers(lifecycle(branch(Component)))>
  </withState(withHandlers(lifecycle(branch(Component))))>
</withState(withState(withHandlers(lifecycle(branch(Component)))))>
`;
